### ISSUE TITLE
possibility: make it unnecessary to pass a function pointer into uiMainSteps()

### ISSUE THREAD

**Issue by andlabs** [andlabs/libui#147]
split from #95 

Actually looking at the GTK+ source code again, I wonder what would happen if I did this

``` diff
diff --git a/unix/main.c b/unix/main.c
index bfd0544..0269d86 100644
--- a/unix/main.c
+++ b/unix/main.c
@@ -41,7 +41,8 @@ int uiMainStep(int wait)
    block = FALSE;
    if (wait)
        block = TRUE;
-   return gtk_main_iteration_do(block) == FALSE;
+   gtk_main_iteration_do(block);
+   return 1;
 }

 // gtk_main_quit() may run immediately, or it may wait for other pending events; "it depends" (thanks mclasen in irc.gimp.net/#gtk+)
```

and ran `uiMainStep()` without `uiMainSteps()`. @gnarz? I could also try this myself I guess... but if this does work I could probably just manage the running flag myself like I do on OS X and avoid the need for a function pointer.

**Comment by gnarz** [comment]
I can/ ill test this, but it is going to take another 1 1/2 weeks until I get back to my linux box.

Von meinem iPhone gesendet

**Comment by andlabs** [comment]
Okay I tested it myself.

It DOES seem to work, however the window positioning and sizing code no longer does (because it depends on `gtk_main()`). I wonder if that can be worked around.

**Comment by andlabs** [comment]
Yep this works, thanks anyway! (Shows what I get for being just a tiny bit lazy =p )

**Comment by andlabs** [comment]
(Copy-pasting from https://github.com/andlabs/libui/pull/21#issuecomment-485597060)

Okay, I need to resurrect this.

>It's useful for any apps where you still want control of the main loop - for instance if you have a game or gamedev tool that uses libui, and you need a tight render-loop and other updates alongside your ui. I guess you could delegate the ui handling to a separate thread, but that seems a bit of a convoluted solution (and needless threads usually cause problems).

How do people do this right now on macOS in a macOS-native application? Because there is no real supported way to do this directly in the same way libui does it now, and even if I did just recreate what `-[NSApplication run]` does internally, that wouldn't be sufficient to replace *all* of `-[NSApplication run]`.

And for the record, I am specifically referring to GUI apps here, not just programs that roll their own poll/kqueue loop.
